### DISCUSSION TITLE
Some question

### DISCUSSION THREAD

**Discussion by xPew** [sn4k3/UVtools#383] [Q&A] [answered]
Hi sn4k3!

I have used UVtools in the past, and now I'd like to revisit it along with my resin printer. There are some new features I'd like to ask about. I have a creality LD002H updated to the latest firmware.
First, when I set up the profiles into SuperSlicer I tried it without making a PrusaSL1 preset, as mentioned in the setup. It worked without it, but I'd like to ask can it cause any trouble, why do you recommend a system preset, and a separate used preset with the modified values? If it is neccesary I will use it, but if possible I'd like to avoid duplicated/unused profiles.
My second question is regarding the new Chitu update. I was glad to see now I can use some of the "smart" features on my LD002H after an update with UVtools. The auto conversion to ctb is set up, but I am not sure if it is already auto converts to v4 or not? I tried to set up FILEVERSION_4 to ensure this, it might take effect but I cant verify. If I manually save to ctbv4 the file extension is v4.ctb, but no matter how and where I place FILEVERSION_4 in the printer profile the autoconversion is only .ctb.
These are my main initial questions, after it, I can play with the new features allowed by this awesome program :)

Thank you for your contribution to the community, and also Merry Christmas :)

**Comment by sn4k3** [comment, marked as answer]
> when I set up the profiles into SuperSlicer I tried it without making a PrusaSL1 preset, as mentioned in the setup. It worked without it, but I'd like to ask can it cause any trouble

It will not.

> why do you recommend a system preset

Because the printers are set as childs of Original Prusa SL1, not really required but it will indicate when original settings were changed. For user this is not required anyway.

> and a separate used preset with the modified values

Because if you use UVtools profile manager to update the base profile or by mistake you update/install other profile with yours selected, it will overwrite and reset all your settings. That's why i recommend. It's good to sync profiles with UVtools changes, sometimes i introduce new variables or fix some values that are really required. But if you manual sync things there aren't any bad in use the base profile.

>  auto converts to v4 or not

UVtools default version for CTB is v3 if not stated the otherwise

> I tried to set up FILEVERSION_4 to ensure this, it might take effect but I cant verify

You can, look at file properties on information tab, and look for "Version" row, also v4 allow TSMC in edit tool parameters, which v3 don't.

> If I manually save to ctbv4 the file extension is v4.ctb, but no matter how and where I place FILEVERSION_4 in the printer profile the autoconversion is only .ctb.

.v4.ctb is a virtual extension, a trick to user and UVtools to force file to v4 when saving, but in the end the extension is .ctb at same, eg: my.file.model.v4.ctb, the extension is still .ctb , only the last dot is valid as extension. So when you set FILEVERSION_4 it will respect and set CTBv4. BTW **FILEVERSION_n** keyword have priority over virtual extension, so if you set FILEVERSION_4 and save as filename.v3.ctb it will set v4, but then if you convert or save the file in the UI it will reencode to v3 latter due the file name. I will drop the virtual extensions at some point

**Reply by xPew** [reply]
Ah, thanks for your insights and explanations. I understand now, and these make sense!

**Comment by xPew** [comment]
One more thing. Does FILEVERSION need to go under the FILEFORMAT before the START_CUSTOM_VALUES, or after it with the other custom values?

**Reply by sn4k3** [reply]
Order doesn't matter as long you don't repeat keywords
